### PR TITLE
refactor!(core): update basic operations of `CMap2`

### PR DESCRIPTION
**Pull request by imrn99**
### Description

**Scope**:  core

**Type of change**: refactor (mostly internals)

**Content description**:

- rename `beta_runtime` to `beta_rt`
- add transactional beta methods
- rewrite cell ID computation to remove all allocations

### Additional information

- [x] Breaking change

### CODE REVIEW

File: honeycomb-core/src/cmap/dim2/basic_ops.rs [hunk 1]
```diff
@@ -235,173 +291,159 @@ impl<T: CoordsFloat> CMap2<T> {
...
+            min = min.min(crt);
+            crt = self.beta_transac::<1>(trans, crt)?;
+        }
+        // if we landed on the null dart, the face is open
+        // we need to iterate in the opposite dir (B0)
+        if crt == NULL_DART_ID {
+            crt = self.beta_transac::<0>(trans, dart_id)?;
+            while crt != NULL_DART_ID {
+                min = min.min(crt);
+                crt = self.beta_transac::<0>(trans, crt)?;
```
[the remark this answers is not in the document]
Comment (imrn99) [reply honeycomb-core/src/cmap/dim2/basic_ops.rs:399]: I replaced these implementations by calling the transactional ones in n `atomically` block. There is a slight difference, but I don't think it matters at this point

File: honeycomb-core/src/cmap/dim2/basic_ops.rs [hunk 2]
```diff
@@ -235,173 +291,159 @@ impl<T: CoordsFloat> CMap2<T> {
...
-        let mut pending = VecDeque::from([dart_id]);
-        while let Some(d) = pending.pop_front() {
-            // WE ASSUME THAT THE FACE IS COMPLETE
-            let image = self.betas[(1, d)].read(trans)?;
-            if marked.insert(image) {
-                // if true, we did not see this dart yet
-                // i.e. we need to visit it later
-                pending.push_back(image);
+        let mut min = dart_id;
+        let mut crt = self.beta_transac::<1>(trans, dart_id)?;
```
[the remark this answers is not in the document]
Comment (imrn99) [reply honeycomb-core/src/cmap/dim2/basic_ops.rs:386]: `crt` the current dart of the traversal. I added a comment

**Comment by imrn99** [comment]
Comments addressed, I have other ides to simplify the sew/link API but I'll make another PR since changes will overflow outside of the core crate too

**Review by cedricchevalier19 — APPROVED** [review]
Thanks for the explanations and the factorization!

**Pull request merged**